crates/red_knot_python_semantic/src/types.rs

```diff
@@ -823,13 +823,18 @@ impl<'db> Type<'db> {
             ),
             (Type::SliceLiteral(..), _) | (_, Type::SliceLiteral(..)) => true,
 
+            (Type::ClassLiteral(..), Type::Instance(InstanceType { class }))
+            | (Type::Instance(InstanceType { class }), Type::ClassLiteral(..)) => {
+                !matches!(class.known(db), Some(KnownClass::Type | KnownClass::Object))
+            }
+            // TODO: function literals are instances of <class 'function'>. Similar for modules
```

Comment:
```suggestion
            // TODO: function literals are instances of types.FunctionType,
            // and modules are instances of types.ModuleType
```

Reply:
These are both variants on the `KnownClass` enum already; it should be pretty trivial to just do the TODO in this PR, if you fancy it?

https://github.com/astral-sh/ruff/blob/953e862aca1449651d96036fcd35f7fab4ccda51/crates/red_knot_python_semantic/src/types.rs#L1507-L1508

Reply:
Oops, for some reason I thought they were not. Of course.